src/plugins/intel_gpu/src/kernel_selector/jitter.cpp

```diff
@@ -259,42 +268,103 @@ class TensorBaseTJitConstant : public JitConstant {
-        {_name + "_PAD_AFTER_SIZE_Y", toCodeString(_tensor.Y().pad.after)},
-        {_name + "_PAD_AFTER_SIZE_Z", toCodeString(_tensor.Z().pad.after)},
-        {_name + "_PAD_AFTER_SIZE_W", toCodeString(_tensor.W().pad.after)},
-        {_name + "_PAD_AFTER_FEATURE_NUM", toCodeString(_tensor.Feature().pad.after)},
-        {_name + "_PAD_AFTER_BATCH_NUM", toCodeString(_tensor.Batch().pad.after)},
-    };
+    size_t idx_offset = _dyn_array_index * 6; // 6D max
+    JitDefinitions definitions{};
+    if (_tensor.is_dynamic()) {
+        auto x = toCodeString(_tensor.X(), idx_offset + 5);
+        auto y = toCodeString(_tensor.Y(), idx_offset + 4);
+        auto z = toCodeString(_tensor.Z(), idx_offset + 3);
+        auto w = toCodeString(_tensor.W(), idx_offset + 2);
+        auto f = toCodeString(_tensor.Feature(), idx_offset + 1);
+        auto b = toCodeString(_tensor.Batch(), idx_offset + 0);
+
+        auto multiply = [](std::vector<std::string> dims) -> std::string {
+            std::string res = "(";
+            for (size_t i = 0; i < dims.size(); i++) {
+                auto& d = dims[i];
+                res += d;
+                if (i != dims.size() - 1)
+                    res += "*";
+            }
+            res += ")";
+            return res;
+        };
+
+        definitions = {
+            {_name + "_SIZE_X", x},
+            {_name + "_SIZE_Y", y},
+            {_name + "_SIZE_Z", z},
+            {_name + "_SIZE_W", w},
+            {_name + "_FEATURE_NUM", f},
+            {_name + "_BATCH_NUM", b},
+            {_name + "_PAD_BEFORE_SIZE_X", toCodeString(_tensor.X().pad.before)},
+            {_name + "_PAD_BEFORE_SIZE_Y", toCodeString(_tensor.Y().pad.before)},
+            {_name + "_PAD_BEFORE_SIZE_Z", toCodeString(_tensor.Z().pad.before)},
+            {_name + "_PAD_BEFORE_SIZE_W", toCodeString(_tensor.W().pad.before)},
+            {_name + "_PAD_BEFORE_FEATURE_NUM", toCodeString(_tensor.Feature().pad.before)},
+            {_name + "_PAD_BEFORE_BATCH_NUM", toCodeString(_tensor.Batch().pad.before)},
+            {_name + "_PAD_AFTER_SIZE_X", toCodeString(_tensor.X().pad.after)},
+            {_name + "_PAD_AFTER_SIZE_Y", toCodeString(_tensor.Y().pad.after)},
+            {_name + "_PAD_AFTER_SIZE_Z", toCodeString(_tensor.Z().pad.after)},
+            {_name + "_PAD_AFTER_SIZE_W", toCodeString(_tensor.W().pad.after)},
+            {_name + "_PAD_AFTER_FEATURE_NUM", toCodeString(_tensor.Feature().pad.after)},
+            {_name + "_PAD_AFTER_BATCH_NUM", toCodeString(_tensor.Batch().pad.after)},
+        };
+        if (_tensor.GetLayout() == DataLayout::bf ||
+            _tensor.GetLayout() == DataLayout::bfyx ||
+            _tensor.GetLayout() == DataLayout::bfzyx ||
+            _tensor.GetLayout() == DataLayout::bfwzyx) {
+            definitions.push_back({_name + "_X_PITCH", "1"});
+            definitions.push_back({_name + "_Y_PITCH", x});
```

Comment:
As I can see, `convert_data_tensor()` function takes into account padding value for pitch calculation, so I think we should care about it in dynamic case as well

Reply:
Fixed